frontend/src/pages/DetailsPage.tsx

```diff
@@ -0,0 +1,35 @@
+import "./DetailsPage.css";
+import {Link, useParams} from "react-router-dom";
+import {TypeDetailsPage, Workout} from "../interfaces/types.ts";
+
+function DetailsPage(props:TypeDetailsPage) {
+    const {id} = useParams();
+    const workout:Workout = props.workouts.filter((workout:Workout):boolean => workout.id === id)[0];
+
+    if(!workout) {
```

Comment:
You can omit the `if` here
```suggestion
    !workout && // <- here then the jsx code
```